system/ns/enter.go

```diff
@@ -15,32 +16,25 @@
 package ns
 
 import (
-	"runtime"
 
 	"github.com/vishvananda/netns"
 )
 
-// NsEnter locks the current goroutine the OS thread and switches to a
-// new network namespace. The returned function must be called in order
-// to restore the previous state and unlock the thread.
+// NsEnter switches to a new network namespace. The returned function must be called in order
+// to restore the previous state
 func Enter(ns netns.NsHandle) (func() error, error) {
-	runtime.LockOSThread()
```

Comment:
I think this has to stay: https://golang.org/pkg/runtime/#LockOSThread
Otherwise it seems that there is no guarantee in which network namespace the Go routine will be.